webapp/src/Service/DOMJudgeService.php

```diff
@@ -419,25 +419,22 @@ public function getUpdates(): array
             }
         }
 
-        if ($this->checkrole('balloon')) {
+        if ($this->checkrole('balloon') && $contest) {
             $balloonsQuery = $this->em->createQueryBuilder()
-            ->select('b.balloonid', 't.name', 't.location', 'p.name AS pname')
-            ->from(Balloon::class, 'b')
-            ->leftJoin('b.submission', 's')
-            ->leftJoin('s.problem', 'p')
-            ->leftJoin('s.contest', 'co')
-            ->leftJoin('p.contest_problems', 'cp', Join::WITH, 'co.cid = cp.contest AND p.probid = cp.problem')
-            ->leftJoin('s.team', 't')
-            ->andWhere('co.cid = :cid')
-            ->andWhere('b.done = 0')
-            ->setParameter('cid', $contest->getCid());
-
-            $freezetime = $contest->getFreezeTime();
-            if ($freezetime !== null && !(bool)$this->config->get('show_balloons_postfreeze')) {
-                $balloonsQuery
-                    ->andWhere('s.submittime < :freeze')
-                    ->setParameter('freeze', $freezetime);
-            }
+                ->select('b.balloonid', 't.name', 't.location', 'p.name AS pname')
+                ->from(Balloon::class, 'b')
+                ->leftJoin('b.submission', 's')
+                ->leftJoin('s.problem', 'p')
+                ->leftJoin('s.contest', 'co')
+                ->leftJoin('p.contest_problems', 'cp', Join::WITH, 'co.cid = cp.contest AND p.probid = cp.problem')
+                ->leftJoin('s.team', 't')
+                ->andWhere('co.cid = :cid')
+                ->setParameter('cid', $contest->getCid())
+                ->andWhere('b.done = 0');
+
+            if (!(bool)$this->config->get('show_balloons_postfreeze')) {
+                $balloonsQuery->andWhere('s.submittime < co.freezetime');
```

Comment:
If there is no freeze, what does this do?

Reply:
That's a good question. I would say we wouldn't handout balloons at all. I'll check for that (or just use the old code again).